src/promptflow/promptflow/_sdk/_telemetry/logging_handler.py

```diff
@@ -73,71 +52,104 @@ def get_scrubbed_cloud_role():
-
-    def __init__(self, custom_properties, enable_telemetry, **kwargs):
+class PromptFlowSDKExporter(AzureMonitorLogExporter):
+    def __init__(self, custom_dimensions, **kwargs):
         super().__init__(**kwargs)
-        # disable AzureEventHandler's logging to avoid warning affect user experience
-        self.disable_telemetry_logger()
-        self._is_telemetry_enabled = enable_telemetry
-        self._custom_dimensions = custom_properties
-
-    def _check_stats_collection(self):
-        # skip checking stats collection since it's time-consuming
-        # according to doc: https://learn.microsoft.com/en-us/azure/azure-monitor/app/statsbeat
-        # it doesn't affect customers' overall monitoring volume
-        return False
-
-    def emit(self, record):
-        # skip logging if telemetry is disabled
-        if not self._is_telemetry_enabled:
-            return
-
-        try:
-            self._queue.put(record, block=False)
+        self._custom_dimensions = custom_dimensions
 
-            # log the record immediately if it is an error
-            if record.exc_info and not all(item is None for item in record.exc_info):
-                self._queue.flush()
-        except Exception:  # pylint: disable=broad-except
-            # ignore any exceptions, telemetry collection errors shouldn't block an operation
-            return
-
-    def log_record_to_envelope(self, record):
+    def _log_to_envelope(self, log_data: LogData) -> TelemetryItem:
         from promptflow._utils.utils import is_in_ci_pipeline
 
-        # skip logging if telemetry is disabled
-
-        if not self._is_telemetry_enabled:
-            return
         custom_dimensions = {
-            "level": record.levelname,
+            "level": log_data.log_record.severity_text,
             # add to distinguish if the log is from ci pipeline
             "from_ci": is_in_ci_pipeline(),
         }
         custom_dimensions.update(self._custom_dimensions)
-        if hasattr(record, "custom_dimensions") and isinstance(record.custom_dimensions, dict):
-            record.custom_dimensions.update(custom_dimensions)
-        else:
-            record.custom_dimensions = custom_dimensions
 
-        envelope = super().log_record_to_envelope(record=record)
+        _attributes = log_data.log_record.attributes
+        if hasattr(_attributes, "custom_dimensions") and isinstance(_attributes.custom_dimensions, dict):
+            custom_dimensions.update(_attributes.custom_dimensions)
+
+        log_data.log_record.attributes = {
+            _APPLICATION_INSIGHTS_EVENT_MARKER_ATTRIBUTE: True,
+            **custom_dimensions,
+        }
+        envelope = super()._log_to_envelope(log_data=log_data)
         # scrub data before sending to appinsights
         role = get_scrubbed_cloud_role()
         envelope.tags["ai.cloud.role"] = role
         envelope.tags.pop("ai.cloud.roleInstance", None)
         envelope.tags.pop("ai.device.id", None)
```

Comment:
what's the typical value for this device.id?

Reply:
![image](https://github.com/microsoft/promptflow/assets/23182548/c9d7577a-4aee-4600-8232-883a8b679ee7)